aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -80,21 +80,46 @@ Tensor logdet(const Tensor& self) {
   return logdet_vals;
 }
 
-std::tuple<Tensor, Tensor> slogdet(const Tensor& self) {
+std::tuple<Tensor, Tensor> linalg_slogdet(const Tensor& self) {
   squareCheckInputs(self);
-  TORCH_CHECK((at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type())),
-              "Expected a floating point tensor as input");
+  TORCH_CHECK((at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type())) && self.dim() >= 2,
```

Comment:
Does this handle bfloat16 and float16 inputs, too?

Reply:
No, this was copied from the old code of `slogdet` and I didn't check what `isFloatingType` does.
I'll change it to accept only float, double, cfloat, cdouble.